include/exadg/postprocessor/pointwise_output_generator_base.h

```diff
@@ -0,0 +1,198 @@
+/*  ______________________________________________________________________
+ *
+ *  ExaDG - High-Order Discontinuous Galerkin for the Exa-Scale
+ *
+ *  Copyright (C) 2021 by the ExaDG authors
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *  ______________________________________________________________________
+ */
+
+#ifndef INCLUDE_COMPRESSIBLE_NAVIER_STOKES_POSTPROCESSOR_POINTWISE_OUTPUT_GENERATOR_BASE_H_
+#define INCLUDE_COMPRESSIBLE_NAVIER_STOKES_POSTPROCESSOR_POINTWISE_OUTPUT_GENERATOR_BASE_H_
+
+#include <deal.II/base/conditional_ostream.h>
+#include <deal.II/base/hdf5.h>
+#include <deal.II/base/mpi_remote_point_evaluation.h>
+#include <deal.II/base/point.h>
+
+#include <deal.II/lac/la_parallel_vector.h>
+
+#include <deal.II/distributed/tria.h>
+
+#include <deal.II/fe/mapping.h>
+
+#include <deal.II/numerics/vector_tools.h>
+
+namespace ExaDG
+{
+template<int dim>
+struct PointwiseOutputDataBase
+{
+  using point_value_type = typename dealii::Point<dim>::value_type;
+
+  PointwiseOutputDataBase();
+
+  bool        write_output;
+  std::string directory;
+  std::string filename;
+
+  bool update_points_before_evaluation;
+
+  double start_time;
+  double end_time;
+  double interval_time;
+
+  std::vector<dealii::Point<dim>> evaluation_points;
+
+  void
+  print(dealii::ConditionalOStream & pcout) const;
+};
+
+template<int dim, typename Number>
+class PointwiseOutputGeneratorBase
+{
+public:
+  using VectorType       = dealii::LinearAlgebra::distributed::Vector<Number>;
+  using point_value_type = typename PointwiseOutputDataBase<dim>::point_value_type;
+
+  PointwiseOutputGeneratorBase(MPI_Comm const & comm);
+
+  virtual ~PointwiseOutputGeneratorBase() = default;
+
+  void
+  setup_base(dealii::DoFHandler<dim> const &      dof_handler_in,
+             dealii::Mapping<dim> const &         mapping_in,
+             PointwiseOutputDataBase<dim> const & pointwise_output_data_in);
+
+  void
+  evaluate(VectorType const & solution, double const & time, int const & time_step_number);
+
+  void
```

Comment:
These functions down to `compute_point_values()` do not have to be public. The public interface should be minimal.